hail/src/main/scala/is/hail/expr/ir/MatrixValue.scala

```diff
@@ -65,7 +65,7 @@ case class MatrixValue(
 
   def stringSampleIds: IndexedSeq[String] = {
     val colKeyTypes = typ.colKeyStruct.types
-    assert(colKeyTypes.length == 1 && colKeyTypes(0).isInstanceOf[TString], colKeyTypes.toSeq)
+    assert(colKeyTypes.length == 1 && colKeyTypes(0) == TString, colKeyTypes.toSeq)
```

Comment:
Would this be better to leave as isInstanceOf, or will we never subclass TString?

Reply:
would have to be `isInstanceOf[TString.type]` which is just weird. Case objects don't define a companion type.

Reply:
ah thanks